src/xrEngine/pure.h

```diff
@@ -98,9 +98,9 @@ class MessageRegistry
             messages[0].Object->OnPure();
         else
         {
-            for (const auto& message : messages)
-                if (message.Prio != REG_PRIORITY_INVALID)
-                    message.Object->OnPure();
+            for (int i = 0; i < messages.size(); i++)
```

Comment:
And looks like the current implementation is probably not ready to modification of the vector size while execution of the cycle. The 'i' variable will be incremented anyway. But if we are modifying the vector, it could lead to skipping some elements or to executing method OnPure twice. Please check it.